components/secure_embed/browser/secure_embed_browsertest.cc

```diff
@@ -423,4 +423,45 @@ IN_PROC_BROWSER_TEST_F(SecureEmbedBrowserTest, ReattachSameGuestToNewEmbed) {
   VerifyBoxRendering(SK_ColorRED);
 }
 
+#if defined(USE_AURA)
+// Test that the proper (top-level) InputManager is used for embedded
+// WebContents. BoundingBoxUpdateWaiter is aura-specific. This triggers
+// a selection change on top-level when the embed is focused.
+IN_PROC_BROWSER_TEST_F(SecureEmbedBrowserTest, InputManager) {
+  auto guest_contents =
+      SetupHarnessAndGuestWithContent("/secure_embed/red_box.html");
+  AttachGuestToEmbed(guest_contents.get());
+
+  content::BoundingBoxUpdateWaiter waiter(web_contents());
+
+  const char kAddTextScript[] = R"(
+     let text = document.createTextNode("Some text to select");
+     document.body.appendChild(text);
+  )";
+  ASSERT_TRUE(content::ExecJs(web_contents(), kAddTextScript));
+  ASSERT_TRUE(content::ExecJs(guest_contents.get(), kAddTextScript));
+
+  const char kFocusScript[] = R"(
+    document.getElementsByTagName("embed")[0].focus();
+  )";
+
+  ASSERT_TRUE(content::ExecJs(web_contents(), kFocusScript));
+
+  const char kSelectionChange[] = R"(
+     let range = document.createRange();
+     range.selectNodeContents(document.body);
+     window.getSelection().addRange(range);
+  )";
+
+  // BoundingBoxUpdateWaiter asks the top-level for active frame's selection
+  // change whenever the top-level's selection changes, so we need to have
+  // both frames change their selections.
+  ASSERT_TRUE(content::ExecJs(guest_contents.get(), kSelectionChange));
+  ASSERT_TRUE(content::ExecJs(web_contents(), kSelectionChange));
+
```

Comment:
The comment above this test says "Test that the proper (top-level) InputManager is used for embedded WebContents", but there is no assertion / expectation on TextInputManager.

Can we use [`TextInputManagerTester`](https://source.chromium.org/chromium/chromium/src/+/main:content/public/test/text_input_test_utils.h;drc=33685ef0a89c1eb9c61f1819dc0029594a53cefb;bpv=1;bpt=1;l=172) to add some explicit expectations? This tester class does not provide direct access to the TextInputManager (due to Manager being a content private thing), but I think can check if the outer and inner WebContents' [`GetActiveView()`](https://source.chromium.org/chromium/chromium/src/+/main:content/public/test/text_input_test_utils.h;l=168;drc=33685ef0a89c1eb9c61f1819dc0029594a53cefb) point to the same View.